### ISSUE TITLE
Preserve return value of `connectWpQuery` query func

### ISSUE THREAD

**Issue by sdgluck** [outlandishideas/kasia#30]
The decorators currently enforce post type-oriented results, by which I mean `connectWpQuery` components receive an object `entities` containing a key for each post type returned by the query, and `connectWpPost` components receive the `[name]` of the post type.

``` js
// connectWpQuery
const { entities: { page, post } } = this.props.kasia

// connectWpPost
const { page } = this.props.kasia
```

This means that complex queries can get out-of-hand. For example, if you wanted to find a "primary article" and its "related articles" you would have to determine within the component which of the `articles` entities is "primary" and which are "related" when you will have already made targeted requests for them in the `queryFn`. So you're doing work twice. :'-(

It would be good if `connectWpQuery` accepted a modifier that made it preserve the return value of its `queryFn` so that we could do stuff like this:

``` js
function queryFn (wpapi) {
  //... do query stuff
  return {
    primaryArticle,
    relatedArticles
  }
}

const { data: { primaryArticle, relatedArticles } } = this.props.kasia
```

API would look something like:

``` js
@connectWpQuery(
  // query function
  query,
  // props comparator
  (a, b) => a.id !== b.id, 
  // we specify to preserve here
  true
)
```

**Comment by sdgluck** [comment]
Thoughts for the implementer...

Points to discuss:
- How should a preserved return value be stored in the state? Inside the query object? (Preserved queries do not get normalised? Or we could implement more sophisticated normalisation...)

Changes to API:
- `connect.js`: `connectWpQuery` should accept a third argument, e.g. `preserveResult`, that when `true` instructs kasia to preserve the return value of the query function.

**Comment by nickpalmer** [comment]
I would advise that if you are going to start adding options to connectWpQuery to make it an options hash as for sure you will find another thing you want an option for quickly. Take using generators as a query function for example. ;)

**Comment by sdgluck** [comment]
@nickpalmer Yep, that is the way we chose to do it. 😄 This feature is implemented on branch `v4.2.0` and available on npm at `kasia@4.2.0-beta.0`.
